### PR TITLE
feat: add chart hooks to migration-job template for app chart

### PR DESCRIPTION
**Pull request by bsushmith**
The current migration-job template does not clean up post the job completion. During app redeployments, this poses an issue, and one has to manually delete the completed migration job.

This change would clear the resource once the migration is done successfully.
Resolves issue - https://github.com/odpf/charts/issues/38

### CODE REVIEW

**Review by AkarshSatija — APPROVED** [review]

File: stable/app/templates/migration-job.yaml [hunk 1]
```diff
@@ -10,7 +10,12 @@ spec:
     metadata:
       {{- with .Values.podAnnotations }}
       annotations:
-        {{- toYaml . | nindent 8 }}
+      # This is what defines this resource as a hook. Without this line, the
```
Comment (AkarshSatija) [inline stable/app/templates/migration-job.yaml:13]: Requires upgrading chart version as well

**Review by AkarshSatija — CHANGES_REQUESTED** [review]

File: stable/app/templates/migration-job.yaml (hunk 1 above)
Comment (bsushmith) [reply stable/app/templates/migration-job.yaml:13]: Done.

File: stable/app/templates/migration-job.yaml [hunk 2]
```diff
@@ -10,7 +10,12 @@ spec:
...
+      # job is considered part of the release.
+      "helm.sh/hook": post-install
+      "helm.sh/hook-weight": "-5"
+      "helm.sh/hook-delete-policy": hook-succeeded
```
Comment (snorlacs) [inline stable/app/templates/migration-job.yaml:17]: This means that only if the job completes successfully we are deleting the job. For failed job we have to manually delete the job always

Comment (snorlacs) [reply stable/app/templates/migration-job.yaml:17]: Can we make hook `pre-install` and delete for both failed and successful jobs

**Review by snorlacs — APPROVED** [review]

**Review by AkarshSatija — APPROVED** [review]

**Pull request merged**